tests/integration/tables/process_open_pipes.cpp

```diff
@@ -0,0 +1,233 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed in accordance with the terms specified in
+ *  the LICENSE file found in the root directory of this source tree.
+ */
+
+// Sanity check integration test for process_open_pipes
+// Spec file: specs/posix/process_open_pipes.table
+
+#include <osquery/logger.h>
+#include <osquery/tests/integration/tables/helper.h>
+
+#include <fcntl.h>
+#include <signal.h>
+#include <sys/types.h>
+#include <unistd.h>
+
+namespace osquery {
+namespace table_tests {
+
+class ProcessOpenPipesTest : public testing::Test {
+  std::string pipe_path;
+  std::string dir_path;
+  std::string test_type;
+  int fd[2];
+  int fd_signal[2];
+
+  void SetUp() override {
+    setUpEnvironment();
+    dir_path = std::string();
+    char dir_template[] = "/tmp/tmpdir.XXXXXX";
+    if (!mkdtemp(dir_template)) {
+      return;
+    }
+    dir_path = std::string(dir_template);
+    pipe_path = dir_path + "/test_pipe";
+    if (mkfifo(pipe_path.c_str(), 0600)) {
+      dir_path = std::string();
+    }
+    if (pipe(fd_signal) == -1) {
+      LOG(ERROR) << "Error creating signal pipe\n";
+    }
+  }
+
+  void TearDown() override {
+    remove(pipe_path.c_str());
+    rmdir(dir_path.c_str());
+    close(fd[0]);
+    close(fd[1]);
+    close(fd_signal[0]);
+    close(fd_signal[1]);
+  }
+
+  void runForever() {
+    while (true) {
+    }
+  }
+  void signal_parent() {
+    char buf = '1';
+    write(fd_signal[1], &buf, 1);
+  }
+
+  int setup_writer() {
+    close(fd_signal[0]);
+    if (test_type == "named_pipe") {
+      int fd = open(pipe_path.c_str(), O_WRONLY);
+      if (fd == -1) {
+        LOG(ERROR) << "Error in opening named pipe";
+      }
+      return fd;
+    } else { // unnamed_pipe
+      close(fd[0]);
+      return fd[1];
+    }
+  }
+
+  int setup_reader() {
+    close(fd_signal[0]);
+    if (test_type == "named_pipe") {
+      int fd = open(pipe_path.c_str(), O_RDONLY);
+      if (fd == -1) {
+        LOG(ERROR) << "Error in opening named pipe";
+      }
+      return fd;
+    } else { // unnamed_pipe
+      close(fd[1]);
+      return fd[0];
+    }
+  }
+
+  void do_writer() {
+    char buf[] = "test";
```

Comment:
nitpick, this is a bit of an unneeded c-ism, this could be a std::string or std::array.